pkg/resolver/errors.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package resolver
+
+import (
+	"fmt"
+	"strings"
+)
+
+// ErrTLDTooLong denotes when a TLD in a name exceeds maximum length.
+type ErrTLDTooLong string
+
+// Error returns the formatted TLD too long error.
+func (e ErrTLDTooLong) Error() string {
+	return fmt.Sprintf("TLD %q exceeds max label length of %d characters", string(e), maxLabelLength)
+}
+
+// ErrInvalidTLD denotes passing an invalid TLD to the MultiResolver.
+type ErrInvalidTLD string
+
+// Error returns the formatted invalid TLD error.
+func (e ErrInvalidTLD) Error() string {
+	return fmt.Sprintf("Invalid TLD %q", string(e))
+}
+
+// ErrResolverChainEmpty denotes trying to pop an empty resolver chain.
+type ErrResolverChainEmpty string
+
+// Error returns the formatted resolver chain empty error.
+func (e ErrResolverChainEmpty) Error() string {
+	return fmt.Sprintf("Resolver chain for %q empty", string(e))
+}
+
+// TODO: implement MultiError as separate package.
```

Comment:
i  feel this may be unnecessary to have this as an error type 
just collect the errors in situ in an array
and return error if non-empty

Reply:
I would consider importing https://github.com/hashicorp/go-multierror, as `node.go` also contains multierror code. However, `node.go` is kind of a mess presently, and it is outside of the scope of this PR in any case.

Reply:
I think that we do not need go-multierror. We can pull put multierror from `node` packge, we just did not need it elsewhere.

Please, write issues for what is required to change in the `node` package, so that we can address them.